### PR TITLE
Add Doxygen for Low Level Documentation

### PR DESCRIPTION
**Pull request by atramirez**
# Overview of changes
This adds Doxygen generated documentation to the project. 
The following are added:
- Doxyfile
- Doxygen Awesome submodule for CSS

## Type of change:

- [X] New Feature

- [ ] Bug Fix

- [X] Documentation

- What is changed in your Pull Request?

## Description

---

This will add doxygen support to allow low level documentation to be read and searched easily. This will generate documentation in a `/docs` that is git ignored. 

## Associated GitHub Issues

---

N/A
## Miscellaneous Information

---

We want to work with DevOps to automate documentation generation, and support a github page.

## Checklist for Review

---

- [X] Continuous Integration (CI) Tests Pass

- [ ] Tests have been created for all files

- [X] Trello Card(s) moved to `Code Review`

- [X] Added `ready for review tag`

- [X] Merge Conflicts Resolved

### CODE REVIEW

**Review by Layendan — DISMISSED** [review]

File: Doxyfile [hunk 1]
```diff
@@ -0,0 +1,2692 @@
...
+# the logo to the output directory.
+
+PROJECT_LOGO           =
+
+# The OUTPUT_DIRECTORY tag is used to specify the (relative or absolute) path
+# into which the generated documentation will be written. If a relative path is
+# entered, it will be relative to the location where doxygen was started. If
+# left blank the current directory will be used.
+
+OUTPUT_DIRECTORY       = "\docs"
```
Comment (tjquillan) [inline Doxyfile:70]: \docs breaks on linux

File: Doxyfile [hunk 2]
```diff
@@ -0,0 +1,2692 @@
...
+# cores available in the system. You can set it explicitly to a value larger
+# than 0 to get more control over the balance between CPU load and processing
+# speed. At this moment only the input processing can be done using multiple
+# threads. Since this is still an experimental feature the default is set to 1,
+# which effectively disables parallel processing. Please report any issues you
+# encounter. Generating dot graphs in parallel is controlled by the
+# DOT_NUM_THREADS setting.
+# Minimum value: 0, maximum value: 32, default value: 1.
+
+NUM_PROC_THREADS       = 1
```
Comment (tjquillan) [inline Doxyfile:487]: Could maybe set this to 0 to enable parallel processing.

**Review by tjquillan — CHANGES_REQUESTED** [review]

File: Doxyfile (hunk 1 above)
Comment (atramirez) [reply Doxyfile:70]: Updating to "docs" this resolves the issue.

File: Doxyfile (hunk 2 above)
Comment (atramirez) [reply Doxyfile:487]: Though we may not need it, this is a good idea for building as the codebase grows, I will update this value to zero.

**Review by tjquillan — APPROVED** [review]

**Review by Layendan — COMMENTED** [review]
Do we need to include the entire doxygen repository in this repository for it to work?

**Comment by atramirez** [comment]
> Do we need to include the entire doxygen repository in this repository for it to work?

No, however Doxygen has been added to the devcontainer build, so everyone should have the ability to build it using `doxygen Doxyfile`. You can view that Pull Request [here](https://github.com/WyroMotion/docker-images/pull/1).  As for the comments themselves we have added the VSCode extension to the devcontianer to allow ease of use.

**Review by Layendan — APPROVED** [review]

**Pull request merged**